app/src/models/banner.ts

```diff
@@ -34,5 +34,5 @@ export type Banner =
       /** name of the branch that was used to rebase */
       readonly targetBranch: string
       /** callback to run when user clicks on link in banner text */
-      readonly onOpenDialog: () => void
+      readonly onOpenDialog: () => Promise<void>
```

Comment:
This change came from the need for an asynchronous callback, and my thinking was that I wanted to make callers aware of this:

https://github.com/desktop/desktop/blob/69c68646d0936ad7320e7935f2ffc5f7af53dbe8/app/src/ui/app.tsx#L1633-L1634

Revisiting this again, based on your additional feedback, I don't see the value in keeping this around. Surfacing a `Promise<void>` here would be a convenient way for callers to be able to know when the asynchronous function completes, but the use case illustrated here is minor and doesn't add value.

> since the showPopup method isn't awaited 

This is an interesting point, because while `AppStore._showPopup` is marked `async` and returning `Promise<void>` (suggesting it should be awaited) there's nothing asynchronous in the function itself that needs to be awaited.

Once I've got some bandwidth after shipping rebase flow I'd like to revisit these sorts of cases and see if we can improve how we're managing void-returning functions and `Promise<void>` returning functions in this area.

Reply:
Reverted in 2824aeb8b917cc5003a18d484f0ef6ce5ec375a9